code/game/machinery/constructable_frame.dm

```diff
@@ -172,7 +172,12 @@
 	else
 		var/option = tgui_input_list(user, "Select Circuitboard To Install"," Available Boards", circuit_boards)
 		target_board = circuit_boards[option]
-		if(QDELETED(target_board) || QDELETED(src) || QDELETED(user) || !(target_board in replacer) || !user.is_holding(replacer) || !user.Adjacent(src))
+		// Everything still where it should be after the UI closed?
+		if(QDELETED(target_board) || QDELETED(src) || QDELETED(user) || !(target_board in replacer) || !user.is_holding(replacer))
```

Comment:
Move this all the way to line 181 outside this else condition so that it applies when he have a single board as well. 

Doesn't make sense we should check for adjacency depending on the number of boards